### ISSUE TITLE
QUESTION: Session problems

### ISSUE THREAD

**Issue by XT3RM1NATOR** [celestix/gotgproto#71]
<img width="625" alt="Screenshot 2024-04-27 at 19 46 17" src="https://github.com/celestix/gotgproto/assets/115317256/0b21abd3-8ec6-4d44-955b-8eb5a292c480">

Even when i pass in a string of a session it still tries to send a code and get a code from me. Additionally it creates the session storage files. you see i need to store the session in the mongodb and how do i properly extract it so that when i create a new client next time the server restarts it wont ask me the code that comes to telegram. 

P.S i extract the session with client.ExportStringSession() maybe there is a better way

**Comment by celestix** [comment]
> <img alt="Screenshot 2024-04-27 at 19 46 17" width="625" src="https://private-user-images.githubusercontent.com/115317256/326185632-0b21abd3-8ec6-4d44-955b-8eb5a292c480.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQyMzI3NzMsIm5iZiI6MTcxNDIzMjQ3MywicGF0aCI6Ii8xMTUzMTcyNTYvMzI2MTg1NjMyLTBiMjFhYmQzLThlYzYtNGQ0NC05NTViLThlYjVhMjkyYzQ4MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyN1QxNTQxMTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03MTAxZjFhMGQyMDI2OTBiYTVlNzEwOGJiNjAwY2Y0M2YwN2I2Y2YzMGU4ZjM5OWRkMTgyNWM2NTg3ZmI2ZWEzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Rw5Wqmib12jft_apLsurLVn1n2snWN4FOjJ892Lmj9s">
> Even when i pass in a string of a session it still tries to send a code and get a code from me. Additionally it creates the session storage files. you see i need to store the session in the mongodb and how do i properly extract it so that when i create a new client next time the server restarts it wont ask me the code that comes to telegram.
> 
> P.S i extract the session with client.ExportStringSession() maybe there is a better way

GoTGProto currently doesn't support storing session in MongoDB, but you can add it yourself if you want. 
Look as https://github.com/celestix/gotgproto/blob/beta/sessionMaker/sessionConstructor.go

You can even make a PR to contribute it for everyone :)

**Comment by XT3RM1NATOR** [comment]
The question is why can i use the string session extracted by client.ExportStringSession() in the format like here Session:         sessionMaker.StringSession() and just save the string however i want

**Comment by celestix** [comment]
What do you mean? Please rephrase your question.

**Comment by XT3RM1NATOR** [comment]
<img width="1200" alt="Screenshot 2024-04-28 at 15 16 06" src="https://github.com/celestix/gotgproto/assets/115317256/b8f90658-b149-4c82-925b-03017718337c">
explained everything here

**Comment by celestix** [comment]
Yeah, you need to use your exported string session as an argument to StringSession function

**Comment by XT3RM1NATOR** [comment]
<img width="881" alt="Screenshot 2024-04-29 at 00 29 55" src="https://github.com/celestix/gotgproto/assets/115317256/6ac4813f-8054-4cc2-aaa8-dad9e4f9895a">

but even when i pass in the exported string session it is still asking me the code as below(the session is fake in the example)

<img width="626" alt="Screenshot 2024-04-29 at 00 30 04" src="https://github.com/celestix/gotgproto/assets/115317256/85090a27-7bd2-4ce2-a46d-b9b6cee0d7da">

and the echo function does not work so the client is not properly set up yet, when using the string session

<img width="633" alt="Screenshot 2024-04-29 at 00 30 16" src="https://github.com/celestix/gotgproto/assets/115317256/1f42ae1c-d9b6-4f64-9db0-d64f8e89d65b">

**Comment by celestix** [comment]
That's strange :thinking: I will check and get back to you.

**Comment by XT3RM1NATOR** [comment]
Thank you very much, keep me updated once you check
)

**Comment by XT3RM1NATOR** [comment]
Any updates?

**Comment by celestix** [comment]
> Any updates?

Please be patient. I am trying to arrange a telegram account for testing it.

**Comment by XT3RM1NATOR** [comment]
Just a quick follow up does the string session work?

**Comment by celestix** [comment]
> Just a quick follow up does the string session work?

Hi, I'm trying to arrange a test account for it but ain't getting luck. Message me on telegram (t.me/coswand), we will figure the way out.

**Comment by celestix** [comment]
> Just a quick follow up does the string session work?

Hi @XT3RM1NATOR, I tried exporting a string session and using it to start the client and it worked completely fine. I think you're doing something wrong at your end, can you show the code you're using to do it?
